myriad-scheduler/src/test/java/com/ebay/myriad/scheduler/fgs/FGSTestBaseSpec.groovy

```diff
@@ -0,0 +1,152 @@
+  EventHandler handler = Mock(EventHandler) {}
+
+  Dispatcher dispatcher = Mock(Dispatcher) {
+    getEventHandler() >> handler
+  }
+
+  RMContext rmContext = Mock(RMContext) {
+    getDispatcher() >> dispatcher
+    getRMApplicationHistoryWriter() >> writer
+    getSystemMetricsPublisher() >> publisher
+    getRMNodes() >> rmNodes
+  }
+
+  /******************* Offers Related ****************/
+
+  Protos.Offer addOfferToFeed(Protos.SlaveID slaveID, String host, int cpu, int mem) {
+    def offer = Protos.Offer.newBuilder()
+        .setId(Protos.OfferID.newBuilder().setValue("test_offer_id"))
+        .setFrameworkId(Protos.FrameworkID.newBuilder().setValue("test_framework_id"))
+        .setSlaveId(slaveID)
+        .setHostname(host)
+        .addResources(Protos.Resource.newBuilder()
+        .setName("cpus")
+        .setScalar(Protos.Value.Scalar.newBuilder().setValue(cpu))
+        .setType(Protos.Value.Type.SCALAR).build())
+        .addResources(Protos.Resource.newBuilder()
+        .setName("mem")
+        .setScalar(Protos.Value.Scalar.newBuilder().setValue(mem))
+        .setType(Protos.Value.Type.SCALAR).build())
+        .build()
+    offerLifecycleManager.addOffers(offer)
+    return offer
+  }
+
+  /******************* Containers Related ****************/
+
+  class FGSContainer {
+    ContainerId containerId
+    Container container
+    RMContainer rmContainer
+    ContainerStatus containerStatus
+  }
+
+  Map<ContainerId, FGSContainer> fgsContainers = new HashMap<>()
+
+  AbstractYarnScheduler yarnScheduler = Mock(AbstractYarnScheduler) {
+    getRMContainer(_ as ContainerId) >> { ContainerId cid -> fgsContainers.get(cid).rmContainer }
+  }
+
+  FGSContainer getFGSContainer(RMNode node, int cid, int cpu, int mem, ContainerState state) {
+    FGSContainer fgsContainer = createFGSContainer(node, cid, cpu, mem, state)
+    if (!fgsContainers.containsKey(fgsContainer.containerId)) {
+      fgsContainers.put(fgsContainer.containerId, fgsContainer)
+    }
+    return fgsContainer
+  }
+
+  private FGSContainer createFGSContainer(RMNode node, int cid, int cpu, int mem, ContainerState state) {
+    ContainerId containerId = ContainerId.newContainerId(ApplicationAttemptId.newInstance(
+        ApplicationId.newInstance(123456789, 1), 1), cid)
+    FGSContainer fgsContainer = new FGSContainer()
+    fgsContainer.containerId = containerId
+    fgsContainer.container = Container.newInstance(containerId, node.getNodeID(), node.getHttpAddress(),
+        Resources.createResource(mem, cpu), null, null)
+    fgsContainer.rmContainer = new RMContainerImpl(fgsContainer.container, containerId.getApplicationAttemptId(),
+        node.getNodeID(), "user1", rmContext)
+    nodeStore.getNode(node.getNodeID().getHost()).getNode().allocateContainer(fgsContainer.rmContainer)
+    fgsContainer.containerStatus = ContainerStatus.newInstance(containerId, state, "", 0);
+    return fgsContainer
+  }
+
+}
```

Comment:
This class scares me... I can't even tell if there is a test in this... It has docs that indicate that it is a base class.   That would likely mean that we are extended for functions which is not good design to me.  I would favor utility classes and helper classes over a base class.  Also the name is too terse.  FGS will not mean anything in 1 year IMO